File: leetcode-problems/two_sum.py
def two_sum(nums,target):
    dic={}
    complement = 0
    listOfIndexes = []
    for i in range(len(nums)-1):
        complement = target - nums[i]
        if(complement in dic):
            listOfIndexes.append(i)
            listOfIndexes.append(dic[complement])
            return listOfIndexes
        else:
            dic[nums[i]]=i
    return -1

listOfNumbers = [2,7,11,15]
print(sorted(two_sum(listOfNumbers,9)))

File: leetcode-problems/two_sum.cpp
#include <algorithm>
#include <iostream>
#include <map>
#include <vector>

using namespace std;

class Solution {
public:
  std::vector<int> twoSum(std::vector<int> &nums, int target) {
    std::vector<int> index;
    std::map<int, int> hashTable;
    int complement = 0;
    for (int i = 0; i < nums.size(); i++) {
      complement = target - nums[i];
      if (hashTable.find(complement) != hashTable.end()) {
        index.push_back(i);
        index.push_back(hashTable[complement]);
      }
      hashTable[nums[i]] = i;
    }
    std::sort(index.begin(), index.end());
    return index;
  }

  void printVector(std::vector<int> &nums) {
    for (int i = 0; i < nums.size(); i++) {
      std::cout << nums[i] << " ";
    }
  }
};

int main() {
  std::vector<int> vec{7, 15, 2, 11};
  auto solved = Solution();
  auto newV = solved.twoSum(vec, 9);
  for (auto &i : newV) {
    std::cout << i << " ";
  }

  return 0;
}